gui/executionGraphGui/client/apps/eg/src/app/model/Socket.ts

```diff
@@ -14,53 +14,20 @@ import * as Long from 'long';
 import { Guid } from 'guid-typescript';
 import { Node } from './Node';
 import { isDefined } from '@eg/common';
-import { Input } from '@angular/compiler/src/core';
 
 export class UIProps {
   public name: string = '';
 }
 
-/**
- * The socket index of a socket `Socket`.
- *
- * @export
- * @class SocketIndex
- * @extends {Long}
- */
 export class SocketIndex extends Long {}
-
-/**
- * Unique Identifier for a `Socket`.
- *
- * @export
- * @class SocketId
- */
-export class SocketId {
-  constructor(private readonly _idString: string = Guid.create().toString()) {}
-  /**
-   * String identifer for this NodeId.
-   *
-   * @returns {string}
-   * @memberof NodeId
-   */
-  public get string(): string {
-    return this._idString;
-  }
-
-  public equal(id: SocketId) {
-    return this._idString === id._idString;
-  }
-}
+export type SocketId = string;
 
 /**
  * Modelclass for a Socket on a node.
- *
- * @export
- * @class Socket
  */
 export abstract class Socket {
```

Comment:
Why not have an interface here? All the logic is only required once, that is where we convert the message to the socket. All other mutations on the model should be done in the reducers. No need for a class here...